parser.js

```diff
@@ -821,27 +827,27 @@
       var from = pos;
       incr(2);
 
-      var enablingFlags = matchReg(/^[sim]+/);
+      var enablingFlags = matchReg(/^([sim]*)([:-])/);
+      if(!enablingFlags){
+        bail('Invalid flags for modifiers group');
+      }
+
       var disablingFlags;
-      if(match("-")){
-        disablingFlags = matchReg(/^[sim]+/);
+      if(enablingFlags[2] === "-") {
+        disablingFlags = matchReg(/^([sim]+):/);
```

Comment:
Aside: Here we disallows the production `(?i-:foo)` when the `enablingFlags` are non-empty and the `disablingFlags` are empty. The current spec draft does allow such productions and the V8 engine supports it well. I have raised a comment in the spec draft review (https://github.com/tc39/ecma262/pull/3221#discussion_r1783354078). If the spec is deemed no change here, I will open a new PR to fix that.